server/controllers/resultsController.js

```diff
@@ -2,46 +2,161 @@ import models from '../models';
+      }
+    } catch (err) {
+      return res.status(500)
+        .json({
+          errors: {
+            message: [err.message]
+          },
+        })
+    }
+  }
+
+  /**
+ * @description - get result for admin and teacher
  * @param {object} req - request object
  * @param {object} res - response object
  * @returns {object} - returns results
  */
- static async addResults(req, res) {
-   try {
-     const { results, duplicates } = res.locals;
-     const uniqueResults = removeDuplicates(results, duplicates);
-     const newResults = await Results
-     .bulkCreate(
-       uniqueResults
-     );
-     if (newResults) {
-       const resObj = {
-         status: 'success',
-         message: `${Object.keys(uniqueResults).length} results successfully added.`
-       }
-       if (Object.keys(duplicates).length) {
-         resObj.duplicates = convertIndexToExcelRow(duplicates);
-       }
-       return res.status(201).json(resObj);
-     }
-   } catch(err){
-     return res.status(500)
-     .json({
-       errors: {
-         message: [err.message]
-       },
-     })
-   }
- }
+  static async getAllResults(req, res) {
+    try {
+      const term = toLowerCase(req.query.term);
+      const year = toLowerCase(req.query.year);
+      const subject = toLowerCase(req.query.subject);
+
+
+      const options = {
+        attributes: ['id', 'year', 'subject', 'exam', 'mark', 'term', 'student_result_id'],
+        where: {},
+        include: [{model: Users, 'as': 'User', attributes: ['user_uid', 'first_name', 'last_name', 'dob', 'year_of_graduation', 'role', 'phone_number', 'email']}],
+        order: [['subject', 'ASC']]
+      };
+
+      options.where.school_uid = req.session.school_uid;
+
+      if (term) {
+        options.where.term = term;
+      }
+
+      if (year) {
+        options.where.year = year;
+      }
+
+      if (subject) {
+        options.where.subject = subject;
+      }
+
+      const results = await Results.findAndCountAll(
+        options
+      )
+
+      return res.status(200).json({
+        status: 'success',
+        message: 'Result retrieved successfully.',
+        results
+      })
+    }
+    catch (err) {
+      return res.status(500)
+        .json({
+          status: 'failure',
+          errors: {
+            message: [err.message]
+          },
+        })
+    }
+  }
+
+  /**
+ * @description - get result for students
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} - returns results
+ */
+  static async getUserResults(req, res) {
+    try {
+      const term = toLowerCase(req.query.term);
+      const year = toLowerCase(req.query.year);
+      const subject = toLowerCase(req.query.subject);
+
+      // specifies options in the findAll sequelize method
+      const options = {
+        attributes: ['id', 'year', 'subject', 'exam', 'mark', 'term', 'student_result_id'],
+        where: {},
+        include: [{model: Users, 'as': 'User'}],
```

Comment:
Should this be the same as `getAllResults`?

`include: [{model: Users, 'as': 'User', attributes: ['user_uid', 'first_name', 'last_name', 'dob', 'year_of_graduation', 'role', 'phone_number', 'email']}]`

Or if this is just retrieving the results of the logged in user, then maybe we don't need to return the user data at all.

What do you think?

Reply:
@jamsgra Just like you suggested, removing the `User` would be logical here.